app/Services/User/OrderService.php

```diff
@@ -28,14 +32,42 @@ public function get(Order $order)
                     ->first();
     }
 
-    public function create($data)
+    public function create($data, array $meals)
     {
         $user = auth()->user();
-
+        $meals = collect($meals);
         $data['required_date'] = now('Europe/London');
 
+        DB::beginTransaction();
+
         $order = $user->orders()->create($data);
 
+        foreach ($meals as $meal) {
+            try {
+                $existing_meal = Meal::findOrFail($meal["meal_id"]);
+                // quantity
```

Comment:
Salaam @afmire877 ,

We need to change 'quantity' here to 'total_quantity_available' otherwise this value does not get updated in the DB.